Civi/Payment/PropertyBag.php

```diff
@@ -804,6 +846,9 @@ public function setFirstName($firstName, $label = 'default') {
    * @return string|null
    */
   public function getInvoiceID($label = 'default') {
+    if (!$this->has('invoiceID')) {
+      $this->set('invoiceID', $label, md5(uniqid(mt_rand(), TRUE)));
+    }
```

Comment:
Add default for invoiceID - generate per CiviCRM core if not defined - makes invoiceID generation consistent if using propertyBag for payment and does not rely on calling code to make sure it has been generated. Calling code can generate if it wants to.

This is a property that is generated by calling code in various places using the same logic. In many cases it is not actually used at all and this reduces coding requirements when submitting payments using propertyBag because it will always have a default if not set manually.